src/services/kpis/kpis.svc.coffee

```diff
@@ -9,19 +9,33 @@ angular
     #====================================
 
     @config = {}
+    @config.userData = {}
 
-    @config.ssoSessionId = ""
     @getSsoSessionId = ->
-      return _self.config.ssoSessionId
+      return _self.config.userData.ssoSessionId
+
+    # @return [Hash] containing user ids important to the kpis & kpis alerting features.
+    #   userId [Integer]
+    #   orgUids [Array]
+    #   ssoSessionId [String]
+    @getUserIds = ->
+      return _self.config.userData
 
     @config.kpisTemplates = []
     @getKpisTemplates = ->
       return _self.config.kpisTemplates
 
-    @config.currentDashboardId = ""
-    @getCurrentDashboardId = ->
-      return _self.config.currentDashboardId
+    @config.currentDhb = {}
+    @getCurrentDashboard = ->
+      return _self.config.currentDhb
+
 
+    #====================================
+    # Register Listeners
+    #====================================
+    ImpacEvents.registerCb(IMPAC_EVENTS.kpiTargetAlert, (notification) ->
+      _self.refreshAll(true)
```

Comment:
you may want to refresh another dashboard (which is not currentDashboard)?
`dashboardId` could be a param included in `notification`, then svc.refreshAll(dashboardId)